native/core/src/parquet/read/values.rs

```diff
@@ -586,8 +586,6 @@ macro_rules! generate_cast_to_unsigned {
     };
 }
 
-generate_cast_to_unsigned!(copy_i32_to_u8, i32, u8, 0_u8);
-generate_cast_to_unsigned!(copy_i32_to_u16, i32, u16, 0_u16);
 generate_cast_to_unsigned!(copy_i32_to_u32, i32, u32, 0_u32);
```

Comment:
Same logic doesn't apply for u32?

Reply:
`copy_i32_to_u32` stays the same as it is for widening `i32` -> `i64(u32)`, so we need to pad zeros